teku/src/main/java/tech/pegasys/teku/cli/options/DepositOptions.java

```diff
@@ -68,17 +69,35 @@ public class DepositOptions {
       fallbackValue = "true")
   private boolean depositSnapshotEnabled = PowchainConfiguration.DEFAULT_DEPOSIT_SNAPSHOT_ENABLED;
 
+  @CommandLine.Spec CommandLine.Model.CommandSpec commandSpec;
+
   public void configure(final TekuConfiguration.Builder builder) {
-    builder.powchain(
-        b ->
-            b.eth1Endpoints(eth1Endpoints)
-                .eth1LogsMaxBlockRange(eth1LogsMaxBlockRange)
-                .useMissingDepositEventLogging(useMissingDepositEventLogging)
-                .depositSnapshotPath(depositSnapshotPath)
-                .depositSnapshotEnabled(parseDepositSnapshotEnabled()));
-  }
+    final CommandLine.ParseResult parseResult = commandSpec.commandLine().getParseResult();
+
+    final String checkpointSyncUrlCliOption = "--checkpoint-sync-url";
+    final boolean checkpointSyncUrlSet = parseResult.hasMatchedOption(checkpointSyncUrlCliOption);
 
-  private boolean parseDepositSnapshotEnabled() {
-    return depositSnapshotEnabled;
+    if (parseResult.hasMatchedOption("--Xdeposit-snapshot")) {
```

Comment:
I think the logic of lines starting from here could be implemented without using low level picocli api. We should have values set here already